workshops/weather_grapher/readme.md

```diff
@@ -0,0 +1,397 @@
+
+We're done setting up the chart!
+
+## Setting up the API   
+We'll be using the Meteostat Web API to get weather data. Before we go to the documentation, head over [here](https://auth.meteostat.net/) to register for an API key, which we'll need when we make requests to the API. Check your email for your API key and save it to your clipboard, we'll use it later.   
+
+[Here](https://dev.meteostat.net/api/#api-key) is the documentation for the API if you want to learn about it more in depth. The basic idea is that the API works by retrieving data made publicly available from various weather stations. So in order to retrieve weather data, we first have to search for weather stations in the in the city we're looking for, and then pick a weather station in that city to retreive data from. First, add the following variables at the very top of your JS file. 
+
+```javascript
+const stationURL = "https://api.meteostat.net/v2/stations/search";
+const dataURL = "https://api.meteostat.net/v2/stations/daily";
+```
+
+`stationURL` is the API endpoint for retrieving various weather stations in an area. `dataURL` is the API endpoint for retreiving weather data from a given weather station. An API endpoint is simply a point of entry, or URL, for the client to receive a specific request from a web API. These URLs can be found in the [documentation](https://dev.meteostat.net/api/#json-api). 
+
+## Retrieving a Weather Station   
+It's time to implement the `getData()` function that we applied to our "Graph" button. Note: All new code segments should be placed directly after the previous code segment unless otherwise specified. Let's start by retreiving the inputs of the city, start date, and end date: 
+
+```javascript 
+async function getData() { // Retrieve the values of city, start date, and end date from input fields. 
+  const city = document.getElementById('city').value;
+  const startDate = document.getElementById('startDate').value;
+  const endDate = document.getElementById('endDate').value;
+}   
+```
+We simply retrieve the values of the inputs using the `.getElementById()` method and the `.value` property. We've made this an `async` function, which means that it will always return a promise. A promise is essentially an object returned by the function to handle a `fetch()` request.     
+
+For our purposes, we use it because it is always required if the `await` expression is used anywhere inside the function, which we'll use in a bit. We also make it an `async` function because then it will pause the function until a promise is returned, which we'll need when making API calls. Read more in-depth about aync/await in javascript [here](https://javascript.info/async-await).    
```

Comment:
```suggestion
For our purposes, we use it because it is always required if the `await` expression is used anywhere inside the function, which we'll use in a bit. We also make it an `async` function because then it will pause the function until a promise is returned, which we'll need when making API calls.

Read more in-depth about aync/await in javascript [here](https://javascript.info/async-await).
```